File: services/backend/core/models/comments.py
from django.utils.timezone import now
from django.db import models
from django.utils.translation import gettext_lazy as _

from .auth import CustomUser
from .products import Product
from .services import Service

# Main Class
class Comment(models.Model):
    description=models.TextField(_("Descripcion"))
    product=models.ForeignKey(Product, verbose_name=_("Producto"), on_delete=models.CASCADE, null=True)
    service=models.ForeignKey(Service, verbose_name=_("Servicio"), on_delete=models.CASCADE, null=True)
    starts=models.IntegerField(_("Estrellas"))
    deleted=models.BooleanField(_("Eliminado"), default=False)
    
    creation_date=models.DateTimeField(_("Fecha Registro"), default=now)
    last_update_date=models.DateTimeField(_("Fecha Actualizacion"), null=True, default=None)
    responsible=models.ForeignKey(CustomUser, verbose_name=_("Responsable"), on_delete=models.CASCADE)
    
    class Meta:
        verbose_name_plural="Comentarios"
    def __str__(self):
        return f"Comentario '{self.description}'"

File: services/frontend/types/dashboard/comments.ts
interface CommentForm{
    description: string,
    product: string,
    service: string,
    starts: number
}

interface CommentItem{
   id: number,
   description: string,
   starts: number,
   creation_date: string,
   product__title: string,
   service__title: string,
   responsible__id: number,
   responsible__username: string,

   deleted: boolean
}

export type {
    CommentItem,
    CommentForm
}